test/unit/io_managers_test.py

```diff
@@ -204,40 +206,53 @@ def test_missing_schema_error(sqlite_io_manager_fixture):
 
 
 @pytest.fixture
-def pudl_sqlite_io_manager_fixture(tmp_path, test_pkg):
-    """Create a SQLiteIOManager fixture with a PUDL database schema."""
-    db_path = tmp_path / "pudl.sqlite"
+def fake_pudl_sqlite_io_manager_fixture(tmp_path, test_pkg, monkeypatch):
+    """Create a SQLiteIOManager fixture with a fake database schema."""
+    db_path = tmp_path / "fake.sqlite"
 
     # Create the database and schemas
     engine = sa.create_engine(f"sqlite:///{db_path}")
     md = test_pkg.to_sql()
     md.create_all(engine)
-    return PudlSQLiteIOManager(base_dir=tmp_path, db_name="pudl", package=test_pkg)
+    return PudlSQLiteIOManager(base_dir=tmp_path, db_name="fake", package=test_pkg)
 
 
-def test_error_when_handling_view_without_metadata(pudl_sqlite_io_manager_fixture):
+def test_migrations_match_metadata(tmp_path, monkeypatch):
+    pkg = Package.from_resource_ids()
+    monkeypatch.chdir(Path(__file__).parent.parent.parent)
+    # alembic knows to use PudlPaths().pudl_db - so we need to set PUDL_OUTPUT env var
+    monkeypatch.setenv("PUDL_OUTPUT", tmp_path)
+    alembic.config.main(["upgrade", "head"])
+
+    PudlSQLiteIOManager(base_dir=tmp_path, db_name="pudl", package=pkg)
```

Comment:
so this guy is running `compare_metadata` which is what is actually doing the checking?? why wasn't this doing it before in the `pudl_sqlite_io_manager_fixture`??

Reply:
or is it `alembic.config.main(["upgrade", "head"])` that is doing the checking? or the combo or the upgrade/hear w/ generating the io manager?